### PR TITLE
Replace solve with solveset in sympy.calculus

### PR DESCRIPTION
**Pull request by aktech**
**TODO**
- [x] `singularities.py`

Partial fix for #8711

@hargup @flacjacket

### CODE REVIEW

File: sympy/calculus/singularities.py [hunk 1]
```diff
@@ -30,4 +31,4 @@ def singularities(expr, sym):
                                   " non rational functions are not yet"
                                   " implemented")
     else:
-        return tuple(sorted(solve(simplify(1/expr), sym)))
+        return tuple(sorted(solveset(simplify(1/expr), sym)))
```
Comment (hargup) [inline sympy/calculus/singularities.py]: We should probably change the output of singularity to sets.

**Comment by hargup** [comment]
+1

File: sympy/calculus/singularities.py (hunk 1 above)
Comment (aktech) [inline sympy/calculus/singularities.py]: Yes, that would be better.

**Comment by moorepants** [comment]
I don't think the solve test should have been removed here.

**Comment by moorepants** [comment]
Ignore my last comment here.

**Pull request merged**